### PR TITLE
Fix build rules for new bootstraper backend

### PR DESCRIPTION
**Pull request by gabrielwen**
Fix some broken rules on both Makefile and Dockerfile

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3754)

### CODE REVIEW

File: bootstrap/cmd/bootstrap/env-setup.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+set -ex
+
+if [ -z ${GOOGLE_APPLICATION_CREDENTIALS} ]; then
+  echo "ENV GOOGLE_APPLICATION_CREDENTIALS not set."
+  exit 1
+fi
+
+PATH=/google-cloud-sdk/bin:/workspace:${PATH}
+
+# Activate service account on bootstrapper.
```
Comment (kunmingg) [inline bootstrap/cmd/bootstrap/env-setup.sh]: Why we need to activate service account here?

File: bootstrap/Dockerfile [hunk 2]
```diff
@@ -86,20 +86,22 @@ ARG registries
 
 COPY $registries /opt/registries
 
-COPY --from=bootstrap_base ${GOPATH}/src/github.com/kubeflow/kubeflow/bootstrap/bin/bootstrapper /opt/kubeflow/
+COPY --from=bootstrap_base /go/src/github.com/kubeflow/kubeflow/bootstrap/bin/bootstrapper /opt/kubeflow/
```
Comment (kunmingg) [inline bootstrap/Dockerfile]: ${GOPATH} should be safer?

File: bootstrap/Dockerfile [hunk 3]
```diff
@@ -47,15 +47,15 @@ COPY . .
 # and allow kfctl to run in a daemon/server mode.
 #
 FROM builder as bootstrap_base
-
+ENV CGO_ENABLED=0
```
Comment (kunmingg) [inline bootstrap/Dockerfile]: Can you comment why the change is necessary?

File: components/gcp-click-to-deploy/ks-app/components/params.libsonnet [hunk 4]
```diff
@@ -3,7 +3,7 @@
   components: {
     ingress: {},
     webapp: {
-      image: "gcr.io/kubeflow-images-public/gcp-click-to-deploy:v20180720-5a864c2c-dirty-f1a621",
+      image: "gcr.io/kubeflow-images-public/gcp-click-to-deploy:v20190725-v0.4.0-rc.1-610-g41443e5e",
```
Comment (kunmingg) [inline components/gcp-click-to-deploy/ks-app/components/params.libsonnet]: I guess it's not been used anywhere?
Maybe remove configs here?

File: bootstrap/Dockerfile (hunk 2 above)
Comment (gabrielwen) [reply bootstrap/Dockerfile]: somehow it could not locate the file:
```
Step 33/44 : COPY --from=bootstrap_base ${GOPATH}/src/github.com/kubeflow/kubeflow/bootstrap/bin/bootstrapper /opt/kubeflow/
COPY failed: stat /usr/local/google/docker/overlay2/2ed96fd138f735c3c95d4a79a3cba2ae81cc5cbd47ae88772743473bf1057b3f/merged/src/github.com/kubeflow/kubeflow/bootstrap/bin/bootstrapper: no such file or directory
```

File: components/gcp-click-to-deploy/ks-app/components/params.libsonnet (hunk 4 above)
Comment (gabrielwen) [reply components/gcp-click-to-deploy/ks-app/components/params.libsonnet]: I intend to.  but let me finish on another part and have another followup PR to remove it for safety.
